scripts/suite_objects.py

```diff
@@ -871,10 +876,20 @@ def match_variable(self, var, run_env):
                 new_vdims = list()
                 new_dict_dims = dict_dims
                 match = True
+            # end if
             # Create compatability object, containing any necessary forward/reverse 
             # transforms from <var> and <dict_var>
             compat_obj = var.compatible(dict_var, run_env)
-
+            # If variable is defined as "inactive" by the host, ensure that
+            # this variable is declared as "optional" by the scheme. If
+            # not satisfied, return error.
+            host_var_active     = dict_var.get_prop_value('active')
+            scheme_var_optional = var.get_prop_value('optional')
+            if (not scheme_var_optional and host_var_active.lower() != '.true.'):
+                errmsg = "Non optional scheme arguments for conditionally allocatable variables"
+                sname  = dict_var.get_prop_value('standard_name')
+                errmsg += ", {}".format(sname)
+                raise CCPPError(errmsg)
```

Comment:
does there also need to be a check to make sure the active attribute is set to something if the variable _is_ optional? Or does it just default to ".true." and that's fine? Maybe a warning?

Reply:
No. If a variable is declared as optional in the scheme metadata, it doesn’t have to be conditionally allocated in the host model (it will just always be allocated and present)